experiments/dsl/src/main/scala/org/apache/gearpump/streaming/dsl/StreamApp.scala

```diff
@@ -100,15 +100,13 @@ class TimeReplayableProducer[T:ClassTag](timeReplayableSource: TimeReplayableSou
 }
 
 class KafkaProducer[T:ClassTag](kafkaConfig: KafkaConfig, converter: Message => T) extends Traverse[T] {
-  val batchSize = kafkaConfig.getConsumerEmitBatchSize
-  val msgDecoder: MessageDecoder = kafkaConfig.getMessageDecoder
-  lazy val source = Some(new KafkaSource(kafkaConfig.getClientId, TaskId(0, 0), 1, kafkaConfig, msgDecoder)).map(kafkaSource => {
-    kafkaSource.startFromBeginning
+  lazy val source = Some(new KafkaSource(kafkaConfig)).map(kafkaSource => {
+    // WARN: the kafka source doesn't work without calling open
     kafkaSource
```

Comment:
DSL currently doesn't work well the existing low-level api so I add a note here and we should address this as follow-ups